### PR TITLE
Bug/d3 asim 3321: Tracking buyer/sellers with their uuids

### PR DESCRIPTION
**Pull request by faizan2590**

### CODE REVIEW

File: src/d3a/models/market/market_structures.py
Comment (faizan2590) [inline src/d3a/models/market/market_structures.py:37]: May be the higher order objects be better now as arguments for classes and methods with too many arguments:
https://stackoverflow.com/questions/5899185/class-with-too-many-parameters-better-design-strategy

Using `dataclasses ` sounds like a good approach for class/method with too many arguments.

**Review by hannesdiedrich — APPROVED** [review]
LGTM

**Comment by hannesdiedrich** [comment]
@faizan2590 Can you please check how much the amount of data in the DB increases with this change? We might have to think about stripping away the uuid parts in the Bids/Offers/Trades.
Also I just thought: is the seller_id really needed? Because as of the current implementation, the seller_name is unique by design (as there are only unique names in one parent area)

**Comment by spyrostz** [comment]
> @faizan2590 Can you please check how much the amount of data in the DB increases with this change? We might have to think about stripping away the uuid parts in the Bids/Offers/Trades.
> Also I just thought: is the seller_id really needed? Because as of the current implementation, the seller_name is unique by design (as there are only unique names in one parent area)

In order to be able to recreate the stats if needed I think we cannot avoid saving the uuids to the DB for the trades. For the bids and the offers though, we can definitely get away by removing the uuids completely.

**Review by spyrostz — DISMISSED** [review]
LGTM

**Comment by faizan2590** [comment]
> > @faizan2590 Can you please check how much the amount of data in the DB increases with this change? We might have to think about stripping away the uuid parts in the Bids/Offers/Trades.
> > Also I just thought: is the seller_id really needed? Because as of the current implementation, the seller_name is unique by design (as there are only unique names in one parent area)
> 
> In order to be able to recreate the stats if needed I think we cannot avoid saving the uuids to the DB for the trades. For the bids and the offers though, we can definitely get away by removing the uuids.

After offline discussion uuid part of Bids/Offers/Trades to be kept in DB for later to decide.

> Also I just thought: is the seller_id really needed? Because as of the current implementation, the seller_name is unique by design (as there are only unique names in one parent area)

Yes they are needed in order to avoid double counting of trade in intermediate hierarchy(https://github.com/gridsingularity/d3a-interface/pull/176/files#diff-4fe53216bd5f83d932a02c99f5062b7714e165902d3aad6274ef33e009b5b2cbR61-R63)

File: integration_tests/steps/integration_tests.py
Comment (spyrostz) [inline integration_tests/steps/integration_tests.py:401]: Please remove these before merging

**Review by spyrostz — DISMISSED** [review]
LGTM

**Review by spyrostz — DISMISSED** [review]
LGTM

File: integration_tests/steps/integration_tests.py
Comment (fievelk) [inline integration_tests/steps/integration_tests.py:395]: If this is not asserting anything and it's just performing some operations, I'd suggest using a `@when` step (and using it in the `.feature` files as `and`).

Comment (fievelk) [reply integration_tests/steps/integration_tests.py:395]: What happens if this step is not performed? If all `test_export_of_kpi_result` must necessarily use this step, I think it would make sense to directly add it inside of `test_export_of_kpi_result`. Please let me know if I'm missing something!

File: src/d3a/models/market/market_structures.py
Comment (fievelk) [reply src/d3a/models/market/market_structures.py:37]: I agree. Just by glancing at it, I have the feeling we could benefit from a `Seller` class to store some of those arguments.

File: src/d3a/d3a_core/rq_job_handler.py
Comment (hannesdiedrich) [inline src/d3a/d3a_core/rq_job_handler.py:26]: please revert

File: integration_tests/kpi.feature
Comment (hannesdiedrich) [inline integration_tests/kpi.feature:7]: Can you please explain, why this is needed?

Comment (hannesdiedrich) [reply integration_tests/kpi.feature:7]: explained offline. thanks

File: integration_tests/steps/integration_tests.py
Comment (hannesdiedrich) [reply integration_tests/steps/integration_tests.py:395]: I agree with Pierpaolo. 
Please add these lines to test_export_of_kpi_result
```
    area_tree_summary = glob.glob(os.path.join(context.export_path, "*", "area_tree_summary.json"))
    with open(area_tree_summary[0], "r") as sf:
        context.area_tree_summary_data = json.load(sf)
```

**Review by hannesdiedrich — DISMISSED** [review]
Two more comments, LGTM once resolved

**Review by spyrostz — DISMISSED** [review]
LGTM

**Review by fievelk — APPROVED** [review]

**Pull request merged**
